internal/biz/model/reporter_resource.go

```diff
@@ -0,0 +1,143 @@
+package model
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// ReporterResourceKey represents the natural key that identifies a resource as reported by a specific reporter.
+// This tuple must be unique across the table.
+type ReporterResourceKey struct {
+	LocalResourceID    string `gorm:"column:local_resource_id;index:reporter_resource_key_idx,unique;index:reporter_resource_search_idx,priority:1;not null;"`
+	ReporterType       string `gorm:"size:128;column:reporter_type;index:reporter_resource_key_idx,unique;index:reporter_resource_search_idx,priority:2;not null;"`
+	ResourceType       string `gorm:"size:128;column:resource_type;index:reporter_resource_key_idx,unique;index:reporter_resource_search_idx,priority:3;not null;"`
+	ReporterInstanceID string `gorm:"size:256;column:reporter_instance_id;index:reporter_resource_key_idx,unique;index:reporter_resource_search_idx,priority:4;not null;"`
+}
+
+// ReporterResource represents the metadata that identifies a resource as reported by a specific reporter.
+// It combines a surrogate UUID primary key with the natural composite key and latest state information.
+type ReporterResource struct {
```

Comment:
This table is normalization of the natural id fields (local_resource_id+resource_type+reporter_type+reporter_instance_id) and partial denormalization of the latest state. Currently we only store the latest version and generation values, but we may also denormalize data in the future, based on query requirements.

This table is also optimized for initial lookup.

This is most similar to Resource table in the legacy model